lib/money/rates_store/memory.rb

```diff
@@ -81,20 +95,32 @@ def transaction(&block)
       # @yieldparam iso_from [String] Currency ISO string.
       # @yieldparam iso_to [String] Currency ISO string.
       # @yieldparam rate [Numeric] Exchange rate.
+      # @yieldparam date [(optional) Date] Date of the exchange rate, only yielded if using a historical store.
       #
       # @return [Enumerator]
       #
       # @example
       #   store.each_rate do |iso_from, iso_to, rate|
       #     puts [iso_from, iso_to, rate].join
       #   end
+      # @example Historical store
+      #   store.each_rate do |iso_from, iso_to, rate, date|
+      #     puts [iso_from, iso_to, rate, date].join
+      #   end
       def each_rate(&block)
         return to_enum(:each_rate) unless block_given?
 
         guard.synchronize do
           rates.each do |key, rate|
-            iso_from, iso_to = key.split(INDEX_KEY_SEPARATOR)
-            yield iso_from, iso_to, rate
+            if @options[:historical]
```

Comment:
How do you feel about using a subclass `RatesStore::MemoryWithHistory` instead of splitting this behaviour?

Reply:
Sure, it would probably be better as a subclass.